rust/main/chains/hyperlane-sealevel/src/provider/recipient/tests.rs

```diff
@@ -0,0 +1,39 @@
+use std::fs;
+use std::path::PathBuf;
+
+use solana_transaction_status::{
+    EncodedConfirmedTransactionWithStatusMeta, EncodedTransactionWithStatusMeta,
+};
+
+use hyperlane_core::H512;
+
+use crate::provider::recipient::RecipientProvider;
+use crate::utils::decode_h256;
+
+#[test]
+fn test() {
+    // given
+    let transaction = transaction(&read_json(
+        "multistep_transaction_including_token_transfer.json",
+    ));
+    let mailbox_address = decode_h256("EitxJuv2iBjsg2d7jVy2LDC1e2zBrx4GB5Y9h2Ko3A9Y").unwrap();
+    let provider = RecipientProvider::new(mailbox_address);
+
+    // when
+    let recipient = provider.recipient(&H512::zero(), &transaction);
+
+    // then
+    assert!(recipient.is_ok());
+    assert_eq!(recipient.unwrap(), mailbox_address);
```

Comment:
I was actually expecting `FJu4E1BDYKVg7aTWdwATZRUvytJZ8ZZ2gQuvPfMWAz9y` to be the "recipient", as this is the warp route program ID. I wouldn't say it's a giant deal but this is the behavior had before (if there weren't other non-native programs invoked) instead of considering the mailbox the recipient. The heuristic could be "search `transaction.message.instructions` for any elements that have the Mailbox in the `accounts`".

But then again, I don't think this is that critical as this address isn't used for anything too important. So if you think it's better to have us just always know that the tx recipient in the scraper db isn't super accurate on SVM, I'm open to it

Reply:
Good point. Fixed